gradio/routes.py

```diff
@@ -144,6 +144,20 @@ def get_blocks(self) -> gradio.Blocks:
             raise ValueError("No Blocks has been configured for this app.")
         return self.blocks
 
+    @staticmethod
```

Comment:
This could just as well not be a `@staticmethod` so it's easy to override in a subclass..?

Reply:
To have an application be able to customize a forwarded request without having to copy-paste as much code.

Reply:
Hmm I'm having a hard time seeing this in practice -- nor have I ever seen the `App` class subclassed in any gradio app. If you have a concrete use case, feel free to share, otherwise I'll keep as is